src/app/ui/question/question.component.ts

```diff
@@ -84,4 +84,12 @@ export class QuestionComponent implements OnInit {
       this.snackBar.open(`Difficulty rating sent Successfully!`, 'close', {duration: 3000});
     });
   }
+
+  removeTag(tag: string) {
+    this.spinner.show();
+    this.db.removeTagFromQuestion(this.qId, tag).then(() => this.spinner.hide()).then(() => {
+      this.snackBar.open(`Removed Tag Successfully!`, 'close', {duration: 3000});
+      this.tags$ = this.db.getTagsOfQuestion(this.qId);
```

Comment:
It is not necessary. The current observable emits a new value after the update, this is the purpose of an observable :)
In fact, I am not sure why there is a separate observable for the tags in the first place. The tags are part of the question itself, so you can remove the tags observable and get the tags from the question itself. You can also delete the getTagsOfQuestion method from db service. Or do I miss something?

Reply:
You are right, my bad. I will fix this.